### DISCUSSION TITLE
(Scout) Frequent MakeSearchable Jobs Dispatched Without Model Updates in Production Environment

### DISCUSSION THREAD

**Discussion by tberk** [laravel/framework#51712] [Q&A]
I am experiencing an issue with Laravel Scout where `MakeSearchable` jobs are being dispatched frequently (every 10-15 seconds) in my production environment, even though there are no updates to the associated model (`Topic`).

In local, there are no such updates everything works as expected.

**Details:**
- Laravel Version: 11.x
- Queue Driver: Redis
- Laravel Horizon is used for queue monitoring
- Deployed to Forge in production.

**Configuration:**
- Only the `Topic` model uses the `Searchable` trait.
```php
 public function toSearchableArray(): array
    {
        return [
            'id'=> $this->id,
            'name'=> $this->name,
            'status' => $this->status,
            'verified' => $this->verified,
            'is_featured' => $this->is_featured,
            'type_id'=> $this->type_id,
            'slug'=> $this->slug,
            'description_short'=> $this->description_short,
            'answer_count' => $this->getAnswerCount(),
            'hits' => $this->getHits(),
        ];
    }  
```

Deploy script on Forge
```
cd /home/forge/*******.com
git pull origin $FORGE_SITE_BRANCH

$FORGE_COMPOSER install --no-dev --no-interaction --prefer-dist --optimize-autoloader

( flock -w 10 9 || exit 1
    echo 'Restarting FPM...'; sudo -S service $FORGE_PHP_FPM reload ) 9>/tmp/fpmlock

if [ -f artisan ]; then
    $FORGE_PHP artisan migrate --force
fi

# Horizon
$FORGE_PHP artisan horizon:terminate

# Meilisearch
$FORGE_PHP artisan scout:sync-index-settings
```

![asd](https://github.com/laravel/framework/assets/11773408/13d10882-c7b7-42a2-aaae-49323b7cf566)

**Comment by wizzymore** [comment]
Hello there, I don't know if you found the issue on your own, but, this happens only in 2 cases:

The model is triggering the saved event (if it was updated or created) or if you trigger the import command for a model.
If you say that the models are not updating then it's the latter, the index is still building up.
You can check this by the parameters passed, it will always be an array but the array will have only one model instance for the saved events and the rest will be chunked and should have the length <= then the one specified in the scout config:
```php
    'chunk' => [
        'searchable' => 500,
        'unsearchable' => 500,
    ],
```

**Comment by tberk** [comment]
Hello, apologies for the delayed response; I was in the midst of moving. Thank you for your detailed reply.

I have not yet identified the issue and have temporarily disabled MeiliSearch. I can confirm that the model is not triggering the saved event. To address the latter possibility, I executed `php artisan scout:import "App\Models\Topic"`, which made all records available. However, this issue has persisted for several days, and I believe that duration is excessively long for index building at that frequency. 🤔
